core/src/smartcontracts/isi/triggers/set.rs

```diff
@@ -211,19 +291,26 @@ impl<'de> DeserializeSeed<'de> for WasmSeed<'_, Set> {
                                 SpecializedAction<ExecuteTriggerEventFilter>,
                             > = map.next_value()?;
                             for (id, action) in triggers {
-                                set.add_by_call_trigger(
-                                    self.loader.engine,
-                                    SpecializedTrigger::new(id, action),
-                                )
-                                .unwrap();
+                                set_transaction
+                                    .add_by_call_trigger(
+                                        self.loader.engine,
+                                        SpecializedTrigger::new(id, action),
+                                    )
+                                    .unwrap();
                             }
                         }
+                        // TODO: ids look redundant because we insert ids already through `add_` methods
```

Comment:
Yeah, a merged field like `triggers: Storage<Id, (EventType, LoadedAction<EventType>)>` might also be possible

Reply:
Created #4575